builds/jazz-build-module/aws-lambda-events-module.groovy

```diff
@@ -138,43 +138,84 @@ def createSqsQueue(queue_name, lambdaARN){
   }
 }
 
-def checkS3AndUpdateLambdaPermissionAndNotification(lambdaARN, s3BucketName, action) {
-  def isExists = checkS3BucketExists(s3BucketName)
-  if(isExists){
-    updateLambdaPermissionAndNotification(lambdaARN, s3BucketName, action)
-  }else{
-    createBucket(s3BucketName)
-    updateLambdaPermissionAndNotification(lambdaARN, s3BucketName, action)
-  }
+def checkS3AndUpdateServerless(s3BucketName, stackName, isEventSchdld) {
+	try {
+		sh "aws s3api head-bucket --bucket $s3BucketName --output json"
+		echo "Bucket exists and have access"
+    checkSlsResourcesAndUpdateServerless(s3BucketName, stackName, isEventSchdld)
+		return true
+	} catch(ex) {//bucket exists but with no access
+		def res
+		try {
+			res = sh (script: "aws s3api head-bucket --bucket $s3BucketName --output json 2<&1 | grep -c 'Forbidden'", returnStdout: true).trim()
+		} catch(e) {
+			echo "Bucket does not exist"
+      enableS3Events()
+			return false;
+		}
+		if(res) {
+			echo "Bucket exists and don't have access"
+			error ex.getMessage()
+		}
+	}
 }
 
-def checkS3BucketExists(s3BucketName){
-   try {
-    sh "aws s3api head-bucket --bucket $s3BucketName --output json"
-    echo "Bucket exists and have access"
-    return true
-  } catch (ex) {//bucket exists but with no access
-    def res
-    try {
-      res = sh(script: "aws s3api head-bucket --bucket $s3BucketName --output json 2<&1 | grep -c 'Forbidden'", returnStdout: true).trim()
-    } catch (e) {
-      echo "Bucket does not exist "
-      return false
+def checkSlsResourcesAndUpdateServerless(s3BucketName, stackName, isEventSchdld){
+  try {
+    def isSlsResource = false
+    def response = sh(
+      script: "aws cloudformation list-stack-resources --stack-name ${stackName} --profile cloud-api --output json",
+      returnStdout: true
+    ).trim()
+    echo "response: $response"
+    def resources = parseJson(response)
+    if(resources.StackResourceSummaries.size() > 0){
+      for(resource in resources.StackResourceSummaries){
+        if(resource.ResourceType == "AWS::S3::Bucket" && resource.PhysicalResourceId == "${s3BucketName}"){
```

Comment:
Just make sure this criteria is sufficient to assume the bucket is created by sls.